qualtran/bloqs/gf_arithmetic/gf2_square_test.py

```diff
@@ -0,0 +1,42 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+from galois import GF
+
+from qualtran.bloqs.gf_arithmetic.gf2_square import _gf2_square_symbolic, _gf16_square, GF2Square
+from qualtran.testing import assert_consistent_classical_action
+
+
+def test_gf16_multiplication(bloq_autotester):
+    bloq_autotester(_gf16_square)
+
+
+def test_gf2_multiplication_symbolic(bloq_autotester):
+    bloq_autotester(_gf2_square_symbolic)
+
+
+def test_gf2_multiplication_classical_sim_quick():
+    m = 2
+    bloq = GF2Square(m)
+    GFM = GF(2**m)
+    assert_consistent_classical_action(bloq, x=GFM.elements)
+
```

Comment:
optional test idea: assert cost is only cliffords

Reply:
done